actions.go

```diff
@@ -3,10 +3,123 @@ package neo4j
 	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
 )
 
+type neo4jAction interface {
+	Run(neo4j.Transaction, map[graph.Node]int64) error
+}
+
+var neo4jActions = map[string]neo4jAction{
+	"Address":     recreate{},
+	"Observation": insert{},
+}
+
+type recreate struct {
+	Config
+	graph.Graph
+	graph.Node
+	entityId int64
+}
+
+func (r recreate) Run(tx neo4j.Transaction, nodeIds map[graph.Node]int64) error {
+	err := loadEntityNodes(tx, r.Graph, []int64{r.entityId}, r.Config, findNeighbors, nil)
+	if err != nil {
+		return err
+	}
+	// delete entity and all nodes reachable from this entity,
+	_, err = tx.Run("MATCH (m)-[*]->(n) WHERE ID(m)=$id DETACH DELETE m,n", map[string]interface{}{"id": r.entityId})
+	if err != nil {
+		return err
+	}
+	// TODO: recreate, find all reachable nodes from this node
+	// create entity
+	vars := make(map[string]interface{})
+	query := fmt.Sprintf("CREATE (m %s %s), return m",
+		r.MakeLabels(r.Node.GetLabels().Slice()),
+		r.MakeProperties(r.Node, vars))
+	idrec, err := tx.Run(query, vars)
+	if err != nil {
+		return err
+	}
+	rec, err := idrec.Single()
+	if err != nil {
+		return err
+	}
+	eid := rec.Values[0].(neo4j.Node).Id
+
+	boundNodes := make(map[graph.Node][]map[string]interface{})
+	batch := make(map[string]graph.Node)
+	queue := make([]graph.Node, 1)
+	queue[0] = r.Node
+	visited := make(map[graph.Node]struct{})
+	for len(queue) > 0 {
+		curr := queue[0]
+		curr.ForEachProperty(func(s string, in interface{}) bool {
+			if _, ok := boundNodes[curr]; ok {
+				boundNodes[curr] = append(boundNodes[curr], map[string]interface{}{r.Config.Map(s): in})
+			} else {
+				boundNodes[curr] = make([]map[string]interface{}, 0)
+			}
+			return true
+		})
+		queue = queue[1:]
+		if _, seen := visited[curr]; !seen {
+			visited[curr] = struct{}{}
+			for edgeItr := curr.GetEdges(graph.OutgoingEdge); edgeItr.Next(); {
+				edge := edgeItr.Edge()
+				if _, seen := visited[edge.GetTo()]; !seen {
+					if _, exists := edge.GetTo().GetProperty(ls.EntitySchemaTerm); !exists {
+						queue = append(queue, edge.GetTo())
+					}
+				}
+			}
+		}
+	}
+
+	query = fmt.Sprintf(`
+		UNWIND $batch as item MATCH (m) where ID(m)=$eid MATCH (n) WHERE 
+		n.$propName IN item CREATE (m)-[%s *]->(n) SET n += item`,
+		r.GetNodes().Node().GetEdges(graph.EdgeDir(graph.OutgoingEdge)).Edge())
+	_, err = tx.Run(query, map[string]interface{}{"batch": batch, "eid": eid, "propName": boundNodes})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type insert struct {
+	Config
+	graph.Node
+	entityId   int64
+	addedNodes []neo4jNode
+}
+
+// insert entity nodes in the db
+// Insert should keep the nodes/edges to be inserted in a list
+func (i insert) Run(tx neo4j.Transaction, nodeIds map[graph.Node]int64) error {
+	vars := make(map[string]interface{})
+	edge := i.GetGraph().GetNodes().Node().GetEdges(graph.EdgeDir(graph.OutgoingEdge)).Edge()
+	query := fmt.Sprintf("MATCH (n) WHERE ID(n) = %d CREATE (n)-[%s %s]->(to %s %s) RETURN to",
```

Comment:
This will create only one node, `to`. What are you trying to do here?

If you are inserting (creating) an entity, build a query containing many create statements.